gems/aws-sdk-s3/spec/object/download_file_spec.rb

```diff
@@ -174,6 +183,53 @@ module S3
               ":chunk_size shouldn't exceed total file size."
             )
         end
+
+        context 'file size is chunk size boundary' do
+          before :each do
+            allow(client).to receive(:head_object).with(
+              bucket: 'bucket',
+              key: 'boundary',
+              part_number: 1
+            ).and_return(
+              client.stub_data(
+                :head_object,
+                content_length: (one_meg * 5) + 1
+              )
+            )
+
+            client.stub_responses(:get_object, ->(context) {
+              file_size = one_meg * 5 + 1
+              # S3 does not allow 1 byte ranges. This shouldn't be raised.
+              if context.params[:range] == "bytes=#{file_size}-#{file_size}"
+                'InvalidRange'
+              else
+                { content_range: "bytes 0-#{one_meg * 5 - 1}/#{file_size}" }
+              end
+            })
+          end
+
+          it 'downloads the file with default chunk size' do
+            expect(client).to receive(:head_object).with(
+              bucket: 'bucket',
+              key: 'boundary',
+              part_number: 1
+            ).exactly(1).times
+
+            boundary_obj.upload_file(boundary_file)
+            boundary_obj.download_file(path)
+          end
+
+          it 'downloads the file with provided chunk size' do
+            expect(client).to receive(:head_object).with(
+              bucket: 'bucket',
+              key: 'boundary',
+              part_number: 1
+            ).exactly(1).times
+
+            boundary_obj.upload_file(boundary_file)
```

Comment:
I know all of the other tests have this pattern as well - but why is it nessecary to call upload_file before download_file.  These aren't integration tests

Reply:
It's not. Just following pattern and easy to reason about.